lib/dao.js

```diff
@@ -2742,6 +2748,7 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
             inst.__persisted = true;
             inst.setAttributes(ctx.data);
+            inst[pkName] = id;
```

Comment:
##### Just to explain and make the review process easier:

You may wonder why I'm not using data which is returned from the connector to make sure data has the `id` which has not been changed, please note that relational databases like Mysql returns something like the following: 

```
OkPacket {
  fieldCount: 0,
  affectedRows: 1,
  insertId: 0,
  serverStatus: 2,
  warningCount: 0,
  message: '(Rows matched: 1  Changed: 1  Warnings: 0',
  protocol41: true,
  changedRows: 1 }
```

So no data is returned for `UPDATE` query.

So thats why I need to set `inset[pkName]` to `id` which is one of the arguments of `replaceById`